File: PDOO/NapakalakiRuby/lib/Treasure.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  class Treasure
    attr_reader :name, :bonus, :type

    def initialize(n, b, t)

      @name = n
      @bonus = b
      @type = t

    end

    def to_s

      "\nNombre: #{@name} \nBonus: #{@bonus} \nTipo: #{@type}"

    end
  end
end

File: PDOO/NapakalakiJava/src/NapakalakiGame/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author pablo
 */
public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind type;
    
    public Treasure( String n, int bon, TreasureKind t){
        name = n;
        bonus = bon;
        type = t;
    }
    public Treasure(){
        
    }
    
    public String getName(){
        
        return name;
        
    }
    
    public int getBonus(){
        
        return bonus;
        
    }
    
    public TreasureKind getType(){
       return type;
    }
    
    public String toString(){
        return "Nombre: " + name + "\nBonus: " + bonus + "\nTipo: " + type;
    }
}
